### DISCUSSION TITLE
Feature Request: Add matrix field view mode buttons

### DISCUSSION THREAD

**Discussion by ryanleichty** [craftcms/cms#15052] [Ideas]
Currently, you can configure the view mode for matrix fields per field, but it would be nice if authors could choose the view mode they prefer when editing content. My suggestion would be to add a button group opposite of the field title that allows you to select the view you want.

Bonus points if this is also configureable per field (select the default view mode, choose whether to show the view mode button group, and choose which view modes are available to select).

![CleanShot 2024-05-22 at 14 44 07@2x](https://github.com/craftcms/cms/assets/17421429/67cedf90-3fc0-4fb4-8219-c1de9b41129d)

**Comment by mmikkel** [comment]
> Bonus points if this is also configureable per field (select the default view mode, choose whether to show the view mode button group, and choose which view modes are available to select).

I'd say that if this happens, a field setting to disable or enable that choice is an absolute **must** 🙂

After working fairly intensively with Craft 5 since its release, my takeaway re: the different view modes for Matrix is that they each tend to lend themselves to different types of content and contexts. _Providing authors with the ability to swap between view modes would essentially strip developers of an important degree of control over the AX we provide to our clients._

I'm sure there are use cases where the view modes could be considered interchangeable, and of course there are going to be power user authors that would appreciate the ability to switch between them, at least for _some_ of their content. But, I can confidently say that for the majority of sites we build, if a feature like this was always-on, it would only serve to make the AX and content model more opaque and would probably become a common source of confusion. 

A lot of people are still getting tripped up with the Structure element index view modes... 🙃

**Reply by thupsi** [reply]
While I like this idea and had thought of pitching it myself, I agree with @mmikkel, this could lead to confusion if always enabled. I had also been thinking of [another way](https://github.com/craftcms/cms/discussions/15053) to address this part:

>different view modes for Matrix each tend to lend themselves to different types of content and contexts.

**Reply by ryanleichty** [reply]
I might've not made it clear, but that's what I meant by the following line:

> choose whether to show the view mode button group

You as the developer should have complete control as to whether or not you want to give the authors the ability to choose their own view mode.

**Reply by thupsi** [reply]
Yes, sorry, I understood your point about this setting being optional. Just wanted to highlight it even more. It was I that was not clear with my wording, so I apologize about that. 🙂
